src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/SafeEvpPKeyHandle.OpenSsl.cs

```diff
@@ -177,5 +193,84 @@ public static SafeEvpPKeyHandle OpenPublicKeyFromEngine(string engineName, strin
+        ///   The URI of the key to open.
+        /// </param>
+        /// <returns>
+        ///   The opened key.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="providerName"/> or <paramref name="keyUri"/> is <see langword="null" />.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   <paramref name="providerName"/> or <paramref name="keyUri"/> is the empty string.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   the key could not be opened via the specified provider.
+        /// </exception>
+        /// <remarks>
+        ///   <para>
+        ///     Both provider name and key URI must be trusted inputs.
+        ///   </para>
+        ///   <para>
+        ///     This operation will fail if OpenSSL cannot successfully load the named <code>provider</code>,
+        ///     or if the named <code>provider</code> cannot load the named key.
+        ///   </para>
+        ///   <para>
+        ///     The syntax for <paramref name="keyUri"/> is determined by each individual
+        ///     <code>provider</code>.
+        ///   </para>
+        /// </remarks>
+        [UnsupportedOSPlatform("android")]
+        [UnsupportedOSPlatform("browser")]
+        [UnsupportedOSPlatform("ios")]
+        [UnsupportedOSPlatform("tvos")]
+        [UnsupportedOSPlatform("windows")]
+        public static SafeEvpPKeyHandle OpenKeyFromProvider(string providerName, string keyUri)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(providerName);
+            ArgumentException.ThrowIfNullOrEmpty(keyUri);
+
+            if (!Interop.OpenSslNoInit.OpenSslIsAvailable)
+            {
+                throw new PlatformNotSupportedException(SR.PlatformNotSupported_CryptographyOpenSSL);
+            }
+
+            return Interop.Crypto.LoadKeyFromProvider(providerName, keyUri);
+        }
+
+        internal int GetKeySizeBits()
+        {
+            return Interop.Crypto.EvpPKeyBits(this);
+        }
+
+        internal int GetKeySizeBytes()
+        {
+            // EVP_PKEY_size returns the maximum suitable size for the output buffers for almost all operations that can be done with the key.
+            // For most of the OpenSSL 'default' provider keys it will return the same size as this method,
+            // but other providers such as 'tpm2' it may return larger size.
+            // Instead we will round up EVP_PKEY_bits result.
+            int keySizeBits = GetKeySizeBits();
+
+            if (keySizeBits <= 0)
+            {
+                Debug.Fail($"EVP_PKEY_bits returned non-positive value: {keySizeBits}");
+                throw new CryptographicException();
+            }
+
+            return (GetKeySizeBits() + 7) / 8;
+        }
+
+        internal Interop.Crypto.EvpAlgorithmId GetKeyType()
+        {
+            return Interop.Crypto.EvpPKeyType(this);
+        }
```

Comment:
I believe the SafeHandle types should only hold SafeHandle state; and therefore GetKeySizeBits and GetKeyType should be deleted (replaced with their direct calls to Interop).  GetKeySizeBytes can be moved to interop to be a pretend interop call.

Not something I'd block on... but something that will almost certainly get changed the next time I have to touch this file.  _Especially_ since this SafeHandle is a public type.